--- a/manifests/compute/rbd.pp
+++ b/manifests/compute/rbd.pp
@@ -77,7 +77,7 @@ class nova::compute::rbd (
 
     exec { 'set-secret-value virsh':
       command => "/usr/bin/virsh secret-set-value --secret $(cat /etc/nova/virsh.secret) --base64 $(ceph auth get-key ${rbd_keyring})",
-      unless  => "/usr/bin/virsh secret-list | grep ${libvirt_rbd_secret_uuid}",
+      unless  => "/usr/bin/virsh secret-get-value ${libvirt_rbd_secret_uuid} | grep $(ceph auth get-key ${rbd_keyring})",
       require => Exec['get-or-set virsh secret']
     }
 

